service/matching/matchingEngine.go

```diff
@@ -211,6 +223,38 @@ func (e *matchingEngineImpl) getTaskListManager(taskList *taskListID,
 	return mgr, nil
 }
 
+func (e *matchingEngineImpl) cacheTaskListByDomainLocked(
```

Comment:
Is it possible to reuse the existing `e.taskLists` field which is basically a cache of all tasklists on the host, instead of creating a new cache? Is the new cache for avoiding lock contention?